src/components/common/ui/form/with-formik/numbered-text-area/FormikNumberedTextArea.tsx

```diff
@@ -0,0 +1,37 @@
+import { FC } from 'react';
+import { useField } from 'formik';
+
+import NumberedTextArea from '../../numbered-text-area-mui/NumberedTextArea';
+import { NumberedTextAreaProps } from '../../numbered-text-area-mui/types';
+import { transformValue } from '../../numbered-text-area-mui/utils';
+
+interface FormikNumberedTextAreaProps
+  extends Omit<NumberedTextAreaProps, 'value'> {
+  name: string;
+}
+
+const FormikNumberedTextArea: FC<FormikNumberedTextAreaProps> = ({
+  name,
+  ...props
+}) => {
+  const [{ value }, { touched, error }, { setValue, setTouched }] =
+    useField(name);
+
+  const onChange = (value: string) => {
+    const transformedValue = transformValue(value);
```

Comment:
You already transform the value inside NumberedTextArea, you don't need to do it twice.